src/linker/Linker/WarningSuppressionWriter.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Mono.Cecil;
+
+namespace Mono.Linker
+{
+	public class WarningSuppressionWriter
+	{
+		private readonly LinkContext _context;
+		private readonly Dictionary<AssemblyNameDefinition, HashSet<(int, IMemberDefinition)>> _warnings;
+
+		public WarningSuppressionWriter (LinkContext context)
+		{
+			_context = context;
+			_warnings = new Dictionary<AssemblyNameDefinition, HashSet<(int, IMemberDefinition)>> ();
+		}
+
+		public void AddWarning (int code, IMemberDefinition memberDefinition)
+		{
+			var assemblyName = _context.Suppressions.GetModuleFromProvider (memberDefinition).Assembly.Name;
+			if (!_warnings.TryGetValue (assemblyName, out var warnings)) {
+				warnings = new HashSet<(int, IMemberDefinition)> ();
+				_warnings.Add (assemblyName, warnings);
+			}
+
+			warnings.Add ((code, memberDefinition));
+		}
+
+		public void OutputSuppressions ()
+		{
+			foreach (var assemblyName in _warnings.Keys) {
```

Comment:
You need to sort (or order) the Keys as well

Reply:
We might, but it's not necessary. The keys are assembly names and we write a separate file for each assembly name. It doesn't matter in which order the files are produced.

Reply:
Right, missed that part